Neos.Media/Classes/Domain/Repository/AssetRepository.php

```diff
@@ -138,7 +147,14 @@ public function countAll()
         $rsm = new ResultSetMapping();
         $rsm->addScalarResult('c', 'c');
 
-        $queryString = "SELECT count(persistence_object_identifier) c FROM neos_media_domain_model_asset WHERE dtype != 'neos_media_imagevariant'";
+        if ($this->entityClassName === Asset::class) {
+            $queryString = 'SELECT count(a.persistence_object_identifier) c FROM neos_media_domain_model_asset a WHERE ' . $this->getAssetVariantFilterClauseForDql('a');
+        } else {
+            $queryString = sprintf(
+                "SELECT count(persistence_object_identifier) c FROM neos_media_domain_model_asset WHERE dtype = '%s'",
```

Comment:
Wait, shouldn't that be `!=` as it was before?

> dtype != 'neos_media_imagevariant'

As said, I don't really follow the logic but just noticed this after looking again

Reply:
No. If this is looking up `Asset` -> do not return any variant, but if looking up some concrete asset type, only that type.

Reply:
Okay, I believe you :)